beacon-chain/rpc/prysm/v2/validator/sync_committee.go

```diff
@@ -42,18 +42,35 @@ func (vs *Server) GetSyncMessageBlockRoot(
 func (vs *Server) SubmitSyncMessage(ctx context.Context, msg *prysmv2.SyncCommitteeMessage) (*emptypb.Empty, error) {
 	errs, ctx := errgroup.WithContext(ctx)
 
+	hState, err := vs.HeadFetcher.HeadState(ctx)
```

Comment:
I can't decide if we should be using HeadState or StatebyRoot here.

My concern is validator's RPC is using some round robin style where request and submit are to different beacon nodes